docs/site/HasMany-relation.md

```diff
@@ -0,0 +1,206 @@
+---
+lang: en
+title: 'Relations'
+keywords: LoopBack 4.0, LoopBack 4
+tags:
+sidebar: lb4_sidebar
+permalink: /doc/en/lb4/HasMany-relation.html
+summary:
+---
+
+## Overview
+
+A `hasMany` relation denotes a one-to-many connection of a model to another
+model through referential integrity. The referential integrity is enforced by a
+foreign key constraint on the target model which usually references a primary
+key on the source model. This relation indicates that each instance of the
+declaring or source model has zero or more instances of the target model. For
+example, in an application with customers and orders, a customer can have many
+orders, as illustrated in the diagram below.
+
+![hasMany relation illustration](./imgs/hasMany-relation-example.png)
+
+The target model, **Order**, has a property, **customerId**, as the foreign key
+to reference the declaring model **Customer's** primary key **id**.
+
+To add a `hasMany` relation to your LoopBack application, you need to perform
+the following steps:
+
+1.  Add a property to your model to access related model instances.
+2.  Modify the source model repository class to provide access to a constrained
+    target model repository.
+3.  Call the constrained target model repository CRUD APIs in your controller
+    methods.
+
+## Defining a hasMany Relation
+
+This section describes how to define a `hasMany` relation at the model level
+using the `@hasMany` decorator. The relation constrains the target repository by
+the foreign key property on its associated model. The following example shows
+how to define a `hasMany` relation on a source model `Customer`.
+
+{% include code-caption.html content="/src/models/customer.model.ts" %}
+
+```ts
+import {Order} from './order.model.ts';
+import {Entity, property, hasmany} from '@loopback/repository';
+
+export class Customer extends Entity {
+  @property({
+    type: 'number',
+    id: true,
+  })
+  id: number;
+
+  @property({
+    type: 'string',
+    required: true,
+  })
+  name: string;
+
+  @hasMany(Order) orders?: Order[];
+
+  constructor(data: Partial<Customer>) {
+    super(data);
+  }
+}
+```
+
+The definition of the `hasMany` relation is inferred by using the `@hasMany`
+decorator. The decorator takes in the target model class constructor and
+optionally a custom foreign key to store the relation metadata. The decorator
+logic also designates the relation type and tries to infer the foreign key on
+the target model (`keyTo` in the relation metadata) to a default value (source
+model name appended with `id` in camel case, same as LoopBack 3). It also calls
+`property.array()` to ensure that the type of the property is inferred properly
+as an array of the target model instances.
+
+The decorated property name is used as the relation name and stored as part of
+the source model definition's relation metadata. The property type metadata is
+also preserved as an array of type `Order` as part of the decoration. Another
```

Comment:
Line break here to separate out the into two paragraphs here please. The last sentence should be reworded as such: `A usage of the decorator ...`